ballerina/tests/associations-tests.bal

```diff
@@ -0,0 +1,293 @@
+
+    WorkspaceInfo expected = {
+        workspaceId: "workspace-22",
+        workspaceType: "medium",
+        location: {
+            buildingCode: "building-22",
+            city: "Manhattan",
+            state: "New York",
+            country: "USA",
+            postalCode: "10570",
+            'type: "owned"
+        },
+        employees: [
+            {
+                empNo: "employee-21",
+                firstName: "Tom",
+                lastName: "Scott",
+                birthDate: {year: 1992, month: 11, day:13},
+                gender: "M",
+                hireDate: {year: 2022, month: 8, day: 1},
+                departmentDeptNo: "department-22",
+                workspaceWorkspaceId: "workspace-22"
+            },
+            {
+                empNo: "employee-22",
+                firstName: "James",
+                lastName: "David",
+                birthDate: {year: 1996, month: 11, day:13},
+                gender: "F",
+                hireDate: {year: 2021, month: 8, day: 1},
+                departmentDeptNo: "department-22",
+                workspaceWorkspaceId: "workspace-22"
+            }
+        ]
+    };
+
+    boolean found = false;
+    _ = from WorkspaceInfo workspace in workspaces
+        do {
+            if (workspace.workspaceId == "workspace-22") {
+                found = true;
+                test:assertEquals(workspace, expected);
+            }
+        };
+
+    if !found {
+        test:assertFail("Expected WorkspaceInfo not found.");
+    }
+
+    check rainierClient.close();
+}
+
+public type BuildingInfo record {|
+    readonly string buildingCode;
+    string city;
+    string state;
+    string country;
+    string postalCode;
+    string 'type;
+    Workspace[] workspaces;
+|};
+
+@test:Config {
+    groups: ["associations"],
+    dependsOn: [employeeRelationsTest]
+}
+function buildingRelationsTest() returns error? {
+    RainierClient rainierClient = check new ();
+
+    stream<BuildingInfo, error?> buildingStream = rainierClient->/building.get();
+    BuildingInfo[] buildings = check from BuildingInfo building in buildingStream 
+        select building;
+
+    BuildingInfo expected = {
+        buildingCode: "building-22",
+        city: "Manhattan",
+        state: "New York",
+        country: "USA",
+        postalCode: "10570",
+        'type: "owned",
+        workspaces: [
+            {
+                workspaceId: "workspace-22",
+                workspaceType: "medium",
+                locationBuildingCode: "building-22"
+            }
+        ]
+    };
+
+    boolean found = false;
+    _ = from BuildingInfo building in buildings
+        do {
+            if (building.buildingCode == "building-22") {
+                found = true;
+                test:assertEquals(building, expected);
+            }
+        };
+
+    if !found {
+        test:assertFail("Expected BuildingInfo not found.");
+    }
+
+    check rainierClient.close();
+
+}
```

Comment:
```suggestion
}

```

Reply:
Fixed in [3ac4df4](https://github.com/ballerina-platform/module-ballerina-persist/pull/155/commits/3ac4df4447e4c445dda487eb932865f44b7f670c)